src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/LicenceSummary.kt

```diff
@@ -23,7 +23,7 @@ data class LicenceSummary(
   val licenceStatus: LicenceStatus,
 
   @Schema(description = "The prison nomis identifier for this offender", example = "A1234AA")
-  val nomisId: String?,
+  val nomisId: String,
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏Are we positive that there will no instances where the nomisId is null? Have a feeling this has come up before - tagging @SebNegahbanMoJ for historical context?

Reply:
There are no existing licences with a null NOMIS ID, and I believe it would likely cause significant issues long before this point if we managed to get one.
There also should be filtering to prevent us being able to create one in the first place, but I'm not sure it's there anymore. I inadvertently removed an explicit check during my frontend refactor, but it was still covered by [this filter](https://github.com/ministryofjustice/create-and-vary-a-licence/blob/main/server/services/lists/caCaseloadService.ts#L274), as `deliusRecord` would be `null` if there were no NOMIS ID.
That doesn't seem to be present anymore ([see here](https://github.com/ministryofjustice/create-and-vary-a-licence-api/pull/556/files#diff-7a111cf0b4dbfa8329f4e11fbbe07c9da52636870204d209e9a59983ca3aac4aR425)) but it may be that Kotlin is doing some automatic null removal?
@nishanthmahasamudram Did the IDE suggest that the filter was unnecessary, hence why it was dropped?

Reply:
Yes, It was IDE suggestion indeed.

Reply:
👍